app/src/main/java/info/nightscout/androidaps/plugins/general/smsCommunicator/SmsCommunicatorPlugin.kt

```diff
@@ -578,12 +579,13 @@ class SmsCommunicatorPlugin @Inject constructor(
             })
         } else if (splitted[1].endsWith("%")) {
             var tempBasalPct = SafeParse.stringToInt(StringUtils.removeEnd(splitted[1], "%"))
+            var durationStep = activePlugin.activePump.model().tbrSettings.durationStep
             var duration = 30
             if (splitted.size > 2) duration = SafeParse.stringToInt(splitted[2])
             val profile = profileFunction.getProfile()
             if (profile == null) sendSMS(Sms(receivedSms.phoneNumber, resourceHelper.gs(R.string.noprofile)))
             else if (tempBasalPct == 0 && splitted[1] != "0%") sendSMS(Sms(receivedSms.phoneNumber, resourceHelper.gs(R.string.wrongformat)))
-            else if (duration == 0) sendSMS(Sms(receivedSms.phoneNumber, resourceHelper.gs(R.string.wrongformat)))
+            else if (duration <= 0 || duration % durationStep != 0) sendSMS(Sms(receivedSms.phoneNumber, String.format(resourceHelper.gs(R.string.wrongTbrDuration), durationStep)))
```

Comment:
`String.format` is not necessary

Reply:
Agreed! I was emulating the style from other instance in this file, e.g. line[878].(https://github.com/nightscout/AndroidAPS/blob/16c0c4cfd3e01ed1326481ea45ee93e6abcadabd/app/src/main/java/info/nightscout/androidaps/plugins/general/smsCommunicator/SmsCommunicatorPlugin.kt#L878)

I suspect this was done to simplify mocking the multi-arg `gs` function, as you can see in `SmsCommunicatorPluginTest` (also for Line 878, which uses the `smscommunicator_tt_set` key). 

Will commit code shortly that also mocks the multi-args `gs` with `mockito.Argumentmatchers`, at least for my code.